openapi-spec.yaml

```diff
@@ -0,0 +1,846 @@
+            type: string
+      tags:
+        - asset management
+      summary: lease an asset
+      description: ""
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                asset:
+                  $ref: "#/components/schemas/Asset"
+      responses:
+        "200":
+          description: "Asset leased successfully"
+  /v1/users/{user_id}/accounts/{account_id}/assets/{asset_id}/asset:pledge:
+    post:
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          description: "The user id"
+          schema:
+            type: string
+        - name: account_id
+          in: path
+          required: true
+          description: "The account id"
+          schema:
+            type: string
+        - name: asset_id
+          in: path
+          required: true
+          description: "The asset id"
+          schema:
+            type: string
+      tags:
+        - asset management
+      summary: pledge an asset
+      description: ""
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                asset:
+                  $ref: "#/components/schemas/Asset"
+      responses:
+        "200":
+          description: "Asset leased successfully"
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: "string"
+          description: "A Unique Account Address to be assigned to the user"
+        pub_key:
+          type: byte
+          description: "The public key the is securely generated on the user side"
+        name:
+          type: "string"
+          description: "The name of the user"
+        email:
+          type: "string"
+          description: "The email of the user"
+    TokenManager:
+      type: object
+      properties:
+        id:
+          type: "string"
+          description: "A Unique Account Address to be assigned to the token manager"
+        name:
+          type: "string"
+          description: "The name of the token manager"
+        email:
+          type: "string"
+          description: "The email of the token manager"
+    Contact:
+      type: object
+    Account:
+      type: object
+    Workflow:
+      type: object
+    Asset:
+      type: object
+      properties:
+        id:
+          type: string
+          description: "A Unique Account Address to be assigned to the asset"
+        allowed_verbs:
+          type: string
+          description: "The allowed verbs for the asset"
+        issuer_id:
+          type: string
+          description: "The issuer of the asset"
+        asset_type:
+          $ref: "#/components/schemas/AssetType"
+        holder_id:
+          oneOf:
+            - type: string
+            - type: array
+              items:
+                type: string
+    SupportedAsset:
```

Comment:
This schema needs to be defined properly. Lets create this schema details in a separate Repo. 
I can send a separate PR for re-org of that.